packages/modular-scripts/src/lint.ts

```diff
@@ -0,0 +1,67 @@
+import * as path from 'path';
+import isCI from 'is-ci';
+
+import { resolveAsBin } from './utils/resolve-as-bin';
+import getModularRoot from './utils/getModularRoot';
+import execSync from './utils/execSync';
+import { getDiffedFiles } from './utils/gitActions';
+import * as logger from './utils/logger';
+
+export interface LintOptions {
+  all: boolean;
+  fix: boolean;
+}
+
+export async function lint(
+  options: LintOptions,
+  regexes: string[] = [],
+): Promise<void> {
+  const { all = false, fix = false } = options;
+  const modularRoot = getModularRoot();
+  const lintExtensions = ['.ts', '.tsx', '.js', '.jsx'];
+  let targetedFiles = ['<rootDir>/**/src/**/*.{js,jsx,ts,tsx}'];
+
+  if (!all && !isCI) {
+    const diffedFiles = getDiffedFiles();
+    if (diffedFiles.length === 0) {
+      logger.log(
+        'No diffed files detected. Add the `--all` option to lint the entire codebase',
+      );
+      return;
+    }
+    targetedFiles = diffedFiles
+      .filter((p: string) => lintExtensions.includes(path.extname(p)))
+      .map((p: string) => `<rootDir>/${p}`);
+  }
+
+  const jestEslintConfig = {
+    runner: require.resolve('modular-scripts/jest-runner-eslint'),
+    displayName: 'lint',
+    rootDir: modularRoot,
+    testMatch: targetedFiles,
+    testPathIgnorePatterns: ['/node_modules/', '/dist/'],
+  };
+
+  const testArgs = [
+    ...regexes,
+    '--config',
+    `"${JSON.stringify(jestEslintConfig)}"`,
+  ];
+
+  const testBin = await resolveAsBin('jest-cli');
+
+  try {
+    execSync(testBin, testArgs, {
+      cwd: modularRoot,
+      log: false,
+      // @ts-ignore
+      env: {
+        MODULAR_ROOT: modularRoot,
+        MODULAR_LINT_FIX: String(fix),
+      },
+    });
+  } catch (_err) {
+    // silence CLI command failure message because jest-runner-eslint handles printing the lint errors
+    process.exit(1);
```

Comment:
Can we hoist this into `cli.ts` which should control process interaction? I think the easiest way would be to return `true` or `false` as to whether an error occurred, or perhaps just the exit code, and exit appropriately.

Reply:
We want to take over the process interaction here because if a test fails, `cli.ts` will print out the entire CLI command with the config as part of the failure. We don't want that. This is similar to how we handle `test.ts` https://github.com/jpmorganchase/modular/blob/main/packages/modular-scripts/src/test.ts#L152